gdnative-derive/src/methods.rs

```diff
@@ -229,9 +269,66 @@ fn impl_gdnative_expose(ast: ItemImpl) -> (ItemImpl, ClassMethodExport) {
+                                                errors.push(syn::Error::new(
+                                                    last.span(),
+                                                    "unexpected type for rpc value, expected Str",
+                                                ));
+                                                return false;
+                                            };
+
+                                            match value.as_str() {
+                                                "remote" => {
+                                                    rpc = RpcMode::Remote;
+                                                    return false;
+                                                }
+                                                "remote_sync" => {
+                                                    rpc = RpcMode::RemoteSync;
+                                                    return false;
+                                                }
+                                                "master" => {
+                                                    rpc = RpcMode::Master;
+                                                    return false;
+                                                }
+                                                "puppet" => {
+                                                    rpc = RpcMode::Puppet;
+                                                    return false;
+                                                }
+                                                "disabled" => {
+                                                    rpc = RpcMode::Disabled;
+                                                    return false;
+                                                }
+                                                "master_sync" => {
+                                                    rpc = RpcMode::MasterSync;
+                                                    return false;
+                                                }
+                                                "puppet_sync" => {
+                                                    rpc = RpcMode::PuppetSync;
+                                                    return false;
+                                                }
+                                                _ => {
+                                                    errors.push(syn::Error::new(
+                                                        last.span(),
+                                                        format!(
+                                                            "unexpected value for rpc: {}",
+                                                            value
+                                                        ),
+                                                    ));
+                                                    return false;
+                                                }
```

Comment:
This can be written a bit more concisely since you return `false` from every branch:

```rust
rpc = match value.as_str() {
    "remote" => RpcMode::Remote,
    "remote_sync" => RpcMode::RemoteSync,
    // - snip --
    _ => {
        errors.push(/* .. */);
        return false;
    },
};
return false;
```

You may also extract this `match` into a `RpcMode::parse(&str) -> Option<Self>` method, which imo is more idiomatic:

```rust
impl RpcMode {
    fn parse(s: &str) -> Option<Self> {
        match s {
            "remote" => Some(RpcMode::Remote),
            "remote_sync" => Some(RpcMode::RemoteSync),
            // - snip -
            _ => None,
        }
    }
}
```

...you can then:

```rust
let mut rpc = None;
// ...
if let Some(mode) = RpcMode::parse(value.as_str()) {
    if rpc.replace(mode).is_some() {
        // Error: RPC mode is already set
    }
} else {
    // Error: unexpected value
}
return false;
// ...
export_args.rpc_mode = rpc.unwrap_or(RpcMode::Disabled);
```

Reply:
That makes sense thanks for the examples